cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -486,6 +497,12 @@ private static File updateHiveConfFile(File confFile, File tempDir) {
     conf.setBoolean(Job.MAPREDUCE_JOB_USER_CLASSPATH_FIRST, false);
     conf.setBoolean(MRJobConfig.MAPREDUCE_JOB_CLASSLOADER, false);
 
+    String sparkHome = System.getenv(Constants.SPARK_HOME);
+    if (sparkHome != null) {
+      LOG.debug("Setting spark.home in hive conf to {}", sparkHome);
+      conf.set("spark.home", sparkHome);
```

Comment:
We can call this an experimental feature in this release. I'll file a JIRA for fixing this in next release.